cv19ResSupportV3.Tests/V3/Gateways/ResidentGatewayTest.cs

```diff
@@ -5,89 +5,477 @@
+        // If Dob is non-empty and not null, then
+        // Resident is considered a duplicate when its Dob, Fname & Lname match.
+        // (Difference between this test and the above one is that for the same given UPRNs, DOBs are no longer empty)
+
+        [TestCase(null)]
+        [TestCase("")]
+        [TestCase(" ")]
+        [TestCase("uprn")]
+        public void FindResidentReturnsAMatchWhenNoNhsAndUprnNumberMatchIsFoundButThereExistsAResidentWithMatchingFirstAndLastNameAndDOBForAllEmptyCasesOfUprn(string testCaseUprn)
+        {
+            // arrange
+            var notMatchingNhsNumber = _faker.Random.Hash();
+            var matchingFirstName = _faker.Random.Hash();
+            var matchingLastName = _faker.Random.Hash();
+            var emptyMatchingUprn = testCaseUprn;
+
+            var matchingDOBDay = _faker.Random.Hash();
+            var matchingDOBMonth = _faker.Random.Hash();
+            var matchingDOBYear = _faker.Random.Hash();
+
+            var searchParameters = new FindResident
+            {
+                NhsNumber = notMatchingNhsNumber,
+                Uprn = emptyMatchingUprn,
+                FirstName = matchingFirstName,
+                LastName = matchingLastName,
+                DobDay = matchingDOBDay,
+                DobMonth = matchingDOBMonth,
+                DobYear = matchingDOBYear
+            };
+
+            var existingResidentNoDOBMatch = new ResidentEntity
+            {
+                NhsNumber = _faker.Random.Hash(),
+                Uprn = _faker.Random.Hash(),
+                FirstName = matchingFirstName,
+                LastName = matchingLastName,
+                DobDay = _faker.Random.Hash(),
+                DobMonth = _faker.Random.Hash(),
+                DobYear = _faker.Random.Hash()
+            };
+
+            var existingResidentDOBMatch = new ResidentEntity
+            {
+                NhsNumber = _faker.Random.Hash(),
+                Uprn = emptyMatchingUprn, // check to see if we fall through to the next rule on empty match
+                FirstName = matchingFirstName,
+                LastName = matchingLastName,
+                DobDay = matchingDOBDay,
+                DobMonth = matchingDOBMonth,
+                DobYear = matchingDOBYear
+            };
+
+            // checking rule falling through to the next on Uprn no match (not just empty)
+            if (emptyMatchingUprn == "uprn") existingResidentDOBMatch.Uprn = _faker.Random.Hash();
+
+            DatabaseContext.ResidentEntities.Add(existingResidentNoDOBMatch);
+            DatabaseContext.ResidentEntities.Add(existingResidentDOBMatch);
+            DatabaseContext.SaveChanges();
+
+            var dobMatchResidentId = DatabaseContext.ResidentEntities
+                .FirstOrDefault(r =>
+                    r.DobDay == matchingDOBDay &&
+                    r.DobMonth == matchingDOBMonth &&
+                    r.DobYear == matchingDOBYear
+                )?.Id;
```

Comment:
Same as above

Reply:
Same answer as for the first 2 cases of this 😁

Reply:
Nvm... adding a record to DB has a side effect that updates the object that you're inserting. Good shout!